pkg/fanal/analyzer/pkg/apk/apk.go

```diff
@@ -112,6 +113,8 @@ func (a alpinePkgAnalyzer) parseApkInfo(scanner *bufio.Scanner) ([]types.Package
 			// e.g. D:scanelf so:libc.musl-x86_64.so.1
 			provides[pkg.Name] = pkg.ID
 		}
+
+		pkg.Identifier = purl.NewPackageIdentifier(types.TargetType(analyzer.TypeApk), pkg)
```

Comment:
i played with your changes and found that you run `purl.NewPackageIdentifier` for each detected line.
We need to call this function after package detection is complete.

Reply:
Do you mean after `parseApkInfo` is invoked? Or even after that?

Reply:
I think we can move this logic before adding `pkg` to `pkgs`:
https://github.com/aquasecurity/trivy/blob/214546427e76da21bbc61a5b70ec00d5b95f6d0b/pkg/fanal/analyzer/pkg/apk/apk.go#L66
https://github.com/aquasecurity/trivy/blob/214546427e76da21bbc61a5b70ec00d5b95f6d0b/pkg/fanal/analyzer/pkg/apk/apk.go#L118

Reply:
Can you also check dpkg and rpm for same problem?

Reply:
done at https://github.com/aquasecurity/trivy/pull/5439/commits/0cb4631731f24cef9f39bf8c5245826de59df005